### PR TITLE
ColorPicker: extend access to presets from gds

### LINKED ISSUE #22411: Add a way to access presets of ColorPicker
**Godot version:**
 c432ce4 

**OS/device including version:**
Xubuntu 16.04

**Issue description:**
While looking at #22402, I found that there is apparently no way for a game to access preset colors of a ColorPicker. You can add your own with `add_preset` but that's it. You can't remove them again by code or get the list of all current presets.
It's also not possible to keep track of presets added by a player as there are no appropriate signals to know when a preset was added or removed from the UI. You _can_ work around the first one by overriding the hidden `_add_preset_pressed` method or finding the "+" button from the picker's internal child nodes at runtime and connect to its `pressed` signal but that seems quite hackish.

### PR DESCRIPTION
**Pull request by lupoDharkael**
this fixes https://github.com/godotengine/godot/issues/22411

new methods:
- void erase_preset(const Color &p_color);
- Array get_presets() const;

new signals:
- "preset_added"
- "preset_removed"

### CODE REVIEW

File: scene/gui/color_picker.cpp [hunk 1]
```diff
@@ -216,12 +216,31 @@ void ColorPicker::add_preset(const Color &p_color) {
...
 	preset->update();
 	if (presets.size() == 10)
 		bt_add_preset->hide();
 }
 
+void ColorPicker::erase_preset(const Color &p_color) {
+
+	if (presets.find(p_color)) {
+		presets.erase(presets.find(p_color));
+		emit_signal("preset_removed", p_color);
```
[the remark this answers is not in the document]
Comment (merumelu) [reply scene/gui/color_picker.cpp]: Agreed!
@lupoDharkael, what I meant in #22411 was that there are no signals when presets are added/removed from the Color Picker GUI. They should be emitted when clicking on the "+" button and when right clicking on a preset to delete it, so somewhere here:
https://github.com/godotengine/godot/blob/6bc18042c2a809a27638f410db577ff95826c49b/scene/gui/color_picker.cpp#L499-L501
https://github.com/godotengine/godot/blob/6bc18042c2a809a27638f410db577ff95826c49b/scene/gui/color_picker.cpp#L444-L449

[the remark this answers is not in the document]
Comment (lupoDharkael) [reply scene/gui/color_picker.cpp]: Ok no problem! I'm chinging it

**Comment by lupoDharkael** [comment]
Now it's fixed, documented in ColorPicker.xml and rebased!

File: scene/gui/color_picker.cpp [hunk 2]
```diff
@@ -222,6 +222,23 @@ void ColorPicker::add_preset(const Color &p_color) {
...
 
+void ColorPicker::erase_preset(const Color &p_color) {
+
+	if (presets.find(p_color)) {
+		presets.erase(presets.find(p_color));
+		preset->update();
+	}
+}
+
+Array ColorPicker::get_presets() const {
```
Comment (merumelu) [inline scene/gui/color_picker.cpp]: Shouldn't this return a `PoolColorArray`?
Not sure if there's an advantage but all the methods I looked at that return single type arrays use `Pool*Array`s if a matching one exists.

Comment (lupoDharkael) [reply scene/gui/color_picker.cpp]: yes, it seems to be a better option. I'm changing it.

**Comment by akien-mga** [comment]
Sorry for the delay, seems fine.

Thanks!

**Pull request merged**